### PR TITLE
docs(samples): add samples for IN, NOT_IN, and != operators.

### LINKED ISSUE #311: Add Datastore query samples for IN, NOT_IN, NOT_EQUAL
Add examples for IN, NOT_IN, and NOT_EQUAL. These samples will also go into the [product documentation](https://cloud.google.com/datastore/docs/concepts/queries)

Depends on #281.

### PR DESCRIPTION
**Pull request by jlara310**
docs: add samples for IN, NOT_IN, and != operators.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-datastore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #311 🦕

### CODE REVIEW

**Comment by jlara310** [comment]
Depends on #287

File: samples/snippets_test.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    def cleanup(self):
+        with self.batch():
+            self.delete_multi(
+                list(set([x.key for x in self.entities_to_delete if x]))
+                + list(set(self.keys_to_delete))
+            )
+
+
+@pytest.yield_fixture
```
Comment (Mariatta) [inline samples/snippets_test.py:39]: I'm seeing `PytestDeprecationWarning` on this line.
Can you please change it to `@pytest.fixture`?

> Warning: @pytest.yield_fixture is deprecated.
Use @pytest.fixture instead; they are the same.
  def client():
```suggestion
@pytest.fixture
```

**Review by Mariatta — CHANGES_REQUESTED** [review]

**Comment by jlara310** [comment]
Now that #287 merged, this is ready for review/merge, too.

**Comment by jlara310** [comment]
The project used for testing requires the indexes in samples/index.yaml. Can someone help me add that index to the testing project?

`gcloud datastore create index.yaml`

**Comment by Mariatta** [comment]
It seems that I also don't have the permission to create the index. I'm trying to find out who could create the index for us.

**Comment by Mariatta** [comment]
The index is now being served, however it doesn't seem like the samples tests are being run as part of the CI.

File: samples/snippets.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from collections import defaultdict
+import datetime
```
Comment (Mariatta) [inline samples/snippets.py:16]: These two lines seems to be not used.

File: samples/snippets.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+from pprint import pprint
+
+import google.cloud.exceptions
+from google.cloud import datastore  # noqa: I100
```
Comment (Mariatta) [inline samples/snippets.py:20]: This is also not used.

File: samples/snippets_test.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+@pytest.mark.flaky
+class TestDatastoreSnippets:
+    # These tests mostly just test the absence of exceptions.
+
+    @backoff.on_exception(backoff.expo, AssertionError, max_time=240)
+    def test_in_query(self, client):
+        tasks = snippets.in_query(client)
+        client.entities_to_delete.extend(tasks)
+        assert tasks
```
Comment (Mariatta) [inline samples/snippets_test.py:54]:
```suggestion
        assert tasks is not None
```

File: samples/snippets.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+def not_equals_query(client):
+    # [START datastore_not_equals_query]
+    query = client.query(kind="Task")
+    query.add_filter("category", "!=", "work")
+    # [END datastore_not_equals_query]
+
+    return list(query.fetch())
+
+def not_equals_query(client):
```
Comment (Mariatta) [inline samples/snippets.py:49]: Duplicated test.

```suggestion
def not_in_query(client):
```

File: samples/snippets_test.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    
+    @backoff.on_exception(backoff.expo, AssertionError, max_time=240)
+    def test_not_equals_query(self, client):
+        tasks = snippets.not_equals_query(client)
+        client.entities_to_delete.extend(tasks)
+        assert tasks
```
Comment (Mariatta) [inline samples/snippets_test.py:60]:
```suggestion
        assert tasks is not None
```

File: samples/snippets_test.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+    @backoff.on_exception(backoff.expo, AssertionError, max_time=240)
+    def test_not_in_query(self, client):
+        tasks = snippets.not_in_query(client)
+        client.entities_to_delete.extend(tasks)
+        assert tasks
```
Comment (Mariatta) [inline samples/snippets_test.py:66]:
```suggestion
        assert tasks is not None
```

File: samples/snippets_test.py [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+    @backoff.on_exception(backoff.expo, AssertionError, max_time=240)
+    def test_eq_query_sorted(self, client):
+        tasks = snippets.eq_query_sorted(client)
+        client.entities_to_delete.extend(tasks)
+        assert tasks
```
Comment (Mariatta) [inline samples/snippets_test.py:72]:
```suggestion
        assert tasks is not None
```

File: samples/snippets_test.py [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+    @backoff.on_exception(backoff.expo, AssertionError, max_time=240)
+    def test_in_query_sorted(self, client):
+        tasks = snippets.in_query_sorted(client)
+        client.entities_to_delete.extend(tasks)
+        assert tasks
```
Comment (Mariatta) [inline samples/snippets_test.py:78]:
```suggestion
        assert tasks is not None
```

**Comment by Mariatta** [comment]
Ok looks like the samples tests are now running, and there are some errors.

**Comment by jlara310** [comment]
Looks like it's just python-docs-samples-tests that is on the legacy back-end. The build-specific projects look to be on Firestore in Datastore mode. Opting into those in the noxfile to use them for testing.

**Comment by jlara310** [comment]
Added a test fixture to set up the required index if it doesn't already exist. However, I'm now getting this error when attempting to create that index:

"google.api_core.exceptions.PermissionDenied: 403 The caller does not have permission"

Either I didn't set things up quite right or the service account doesn't have index creation permissions on the build-specific test projects.

**Comment by jlara310** [comment]
I removed the snippets that require an index for now. I'd like to submit the remaining snippets as that will allow me to update the documentation. I can figure out the other two snippets in a follow-up PR.

I'm still failing two checks, but I'm not sure why. Any insights, appreciated.

Kokoro — Build failed error:

nox > * docs: failed

Kokoro Prerelease Dependencies error:

ERROR collecting samples/snippets/snippets_test.py 
Traceback:
/usr/local/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
samples/snippets/snippets_test.py:16: in <module>
    import backoff
E   ModuleNotFoundError: No module named 'backoff'

**Comment by jlara310** [comment]
PR should be good to go and ready to merge. Thank you.

File: samples/snippets/snippets.py [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+
+def main(project_id):
+    client = datastore.Client(project_id)
+
+    for name, function in globals().items():
+        if name in ("main", "_preamble", "defaultdict") or not callable(function):
+            continue
+
+        print(name)
```
Comment (Mariatta) [inline samples/snippets/snippets.py:66]: Just wondering what is the purpose of the prints here?

One of the output is for the `pprint`:

```
pprint
<google.cloud.datastore.client.Client object at 0x1096a7820>
None
```

Perhaps we don't need these prints?

**Review by Mariatta — APPROVED** [review]

**Comment by Mariatta** [comment]
The pre-release dependencies CI is not needed so it's good to merge. Thank you!

**Pull request merged**